langkit/dsl_unparse.py

```diff
@@ -1068,11 +1068,24 @@ def emit_doc(doc):
 def emit_prop(prop, walker):
     quals = ""
 
-    if prop.is_public:
+    # When a property declared in the DSL happens to be the root of a property
+    # dispatching tree, it is hidden by an artificial property (the
+    # "dispatcher", see CompileCtx.lower_properties_dispatching). The actual
+    # "is_public" and abstract expression for this root property are still
+    # attached to the dispatcher.
+    assert not prop.is_artificial_dispatcher
+    dispatcher = prop.dispatcher
+    prop_for_walker = (dispatcher
+                       if dispatcher and not prop.base
+                       else prop)
+
+    if prop_for_walker.is_public:
         quals += "@export "
 
-    if prop.abstract:
-        quals += "abstract "
+    if not prop.constructed_expr:
+        quals += "@abstract "
+        if prop.abstract_runtime_check:
+            quals += "@abstract_stub "
```

Comment:
I'm not fond of that name, even though I acknowledge the need to find something better than `abstract_runtime_check`. What about `runtime_error` or `not_implemented` ?

```ocaml
@not_implemented fun foo()
```